packages/react-hooks/src/useOnEvent.ts

```diff
@@ -16,17 +16,19 @@ export function useOnEvent<TElement extends Element, TEvent extends Event>(
 ) {
   // Use a ref for the callback to prevent repeatedly attaching/unattaching callbacks that are unstable across renders
   const callbackRef = React.useRef(callback);
-  callbackRef.current = callback;
 
   React.useEffect(() => {
-    if (element && 'current' in element) {
-      element = element.current;
-    }
-    if (!element) {
+    callbackRef.current = callback;
```

Comment:
Oops, real issue: it's unsafe to write to `.current` during render. See React's examples: https://reactjs.org/docs/hooks-faq.html#how-to-read-an-often-changing-value-from-usecallback

Reply:
I see that page wrapping the write to `.current` in `useEffect`, but I don't see anything explaining why writing to it during render is wrong? And if that is a problem, we need to fix it in a lot of places and should add a lint rule.

Reply:
Is the problem related to concurrent mode? (The page vaguely mentions this but doesn't go into detail.) I guess the analogous concurrent mode problem from class components would be that you should be careful about mutating state during render, since a render method may be called multiple times. @MLoughry do you have any thoughts on this one?

Reply:
I don't see in the docs where it says that writing to `.current` during render is unsafe; it's just an unobserved mutable object. Regardless, I think the change is benign.

Reply:
Yes, it's about concurrent mode:

https://medium.com/better-programming/how-to-properly-use-the-react-useref-hook-in-concurrent-mode-38c54543857b

Reply:
Ugh...we are already doing this wrong in a LOT of places and it's going to be a big headache to fix. :( I'm surprised the hooks lint rules don't try to cover this...

Reply:
Reading that article, it's less about "never assign to current during render", and more "realize that render may be run multiple times per commit".

For example, our implementation of `useConst` should be fine, even in concurrent mode.

Reply:
Hmm yeah I guess it might not be as bad as I thought--I'd thought `usePrevious` was doing this wrong, but actually it's fine. But I will still do a search through the code and check for any instances of where this isn't okay, and try to make the rest of the team aware of this issue.